### PR TITLE
doc update core:num class

### PR DESCRIPTION
**Pull request by jpsiitonen**
Added few examples to num class. Any ideas how to improve documentation?

### CODE REVIEW

File: sdk/lib/core/num.dart [hunk 1]
```diff
@@ -184,6 +191,13 @@ abstract class num implements Comparable<num> {
   /// The result is an [int], as described by [int.remainder],
   /// if both this number and [other] is an integer,
   /// otherwise the result is a [double].
+  ///
+  /// Example:
+  /// ```dart
+  /// print(3.remainder(1)); // 0
+  /// print(10.remainder(3)); // 1
+  /// print(5.remainder(3.0)); // 2.0
```
Comment (alexlindroos) [inline sdk/lib/core/num.dart:200]: dartpad gives output of `2`

File: sdk/lib/core/num.dart [hunk 2]
```diff
@@ -377,6 +396,14 @@ abstract class num implements Comparable<num> {
   ///
   /// The arguments [lowerLimit] and [upperLimit] must form a valid range where
   /// `lowerLimit.compareTo(upperLimit) <= 0`.
+  ///
+  /// Example:
+  /// ```dart
+  /// var result = 10.5.clamp(5, 10.0); // 10.0
```
Comment (alexlindroos) [inline sdk/lib/core/num.dart:403]: Dartpad gives output as `10`

File: sdk/lib/core/num.dart [hunk 3]
```diff
@@ -377,6 +396,14 @@ abstract class num implements Comparable<num> {
...
+  /// result = 0.75.clamp(5, 10.0); // 5
+  /// result = (-10).clamp(-5, 5.0); // -5
+  /// result = (-0).clamp(-5, 5.0); // 0
```
Comment (alexlindroos) [inline sdk/lib/core/num.dart:406]: Dartpad gives output as `-0.0`

File: sdk/lib/core/num.dart [hunk 4]
```diff
@@ -517,6 +544,22 @@ abstract class num implements Comparable<num> {
...
   /// you should use `num.tryParse(string) ?? (...)`.
+  ///
+  /// Examples:
+  /// ```dart
+  /// var value = num.parse('2021'); // 2021
+  /// value = num.parse('3.14'); // 3.14
+  /// value = num.parse('  3.14 \xA0'); // 3.14
+  /// value = num.parse('0.'); // 0.0
+  /// value = num.parse('.0'); // 0.0
+  /// value = num.parse('-1.e3'); // -1000.0
```
Comment (alexlindroos) [inline sdk/lib/core/num.dart:556]: Dartpad gives `-1000`

File: sdk/lib/core/num.dart [hunk 5]
```diff
@@ -517,6 +544,22 @@ abstract class num implements Comparable<num> {
   /// The [onError] parameter is deprecated and will be removed.
   /// Instead of `num.parse(string, (string) { ... })`,
   /// you should use `num.tryParse(string) ?? (...)`.
+  ///
+  /// Examples:
+  /// ```dart
+  /// var value = num.parse('2021'); // 2021
+  /// value = num.parse('3.14'); // 3.14
+  /// value = num.parse('  3.14 \xA0'); // 3.14
+  /// value = num.parse('0.'); // 0.0
```
Comment (alexlindroos) [inline sdk/lib/core/num.dart:554]: Dartpad gives `0`

File: sdk/lib/core/num.dart [hunk 6]
```diff
@@ -517,6 +544,22 @@ abstract class num implements Comparable<num> {
...
+  /// value = num.parse('.0'); // 0.0
```
Comment (alexlindroos) [inline sdk/lib/core/num.dart:555]: Dartpad gives `0`

File: sdk/lib/core/num.dart [hunk 7]
```diff
@@ -528,6 +571,22 @@ abstract class num implements Comparable<num> {
   ///
   /// Like [parse] except that this function returns `null` for invalid inputs
   /// instead of throwing.
+  ///
+  /// Examples:
+  /// ```dart
+  /// var value = num.tryParse('2021'); // 2021
+  /// value = num.tryParse('3.14'); // 3.14
+  /// value = num.tryParse('  3.14 \xA0'); // 3.14
+  /// value = num.tryParse('0.'); // 0.0
```
Comment (alexlindroos) [inline sdk/lib/core/num.dart:581]: Dartpad gives `0`

File: sdk/lib/core/num.dart [hunk 8]
```diff
@@ -528,6 +571,22 @@ abstract class num implements Comparable<num> {
...
+  /// value = num.tryParse('.0'); // 0.0
```
Comment (alexlindroos) [inline sdk/lib/core/num.dart:582]: Dartpad gives `0`

File: sdk/lib/core/num.dart [hunk 9]
```diff
@@ -528,6 +571,22 @@ abstract class num implements Comparable<num> {
...
+  /// value = num.tryParse('-1.e3'); // -1000.0
```
Comment (alexlindroos) [inline sdk/lib/core/num.dart:583]: Dartpad gives `-1000`

**Review by stepthomp — COMMENTED** [review]
Twice there is the sentence "Parses a string containing a number literal into a number". This doesn't sound right to me but I'm not sure how it can be fixed to retain the correct meaning.

**Pull request closed without merging**